api/Updated-Metrics.md

```diff
@@ -0,0 +1,368 @@
+There is little advantage to the change at this point.
+This will change in the next section.
+
+## Obtaining Scalars
+
+### Existing Syntax
+
+We provide methods for turning the `Bunch`es returned from `group_summary()` into scalars:
+```python
+>>> difference_from_summary(result)
+0.4406
+>>> ratio_from_summary(result)
+0.3259
+>>> group_max_from_summary(result)
+0.6536
+>>> group_min_from_summary(result)
+0.2130
+```
+We also provide wrappers such as `accuracy_score_difference()`, `accuracy_score_ratio()` and `accuracy_score_min()` for user convenience.
+
+One point which these helpers lack (although it could be added) is the ability to select alternative values for measuring the difference and ratio.
+For example, the user might not be interested in the difference between the maximum and minimum, but the difference from the overall value.
+Or perhaps the difference from a particular group.
+
+### Proposed Change
+
+The `GroupedMetric` object would have methods for calculating the required scalars.
+First, let us consider the differences.
+
+We would provide operations to calculate differences in various ways (all of these results are a Pandas Series):
+```python
+>>> result.differences()
+Male      0.0
+Female    0.4406
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min')
+Male     -0.4406
+Female    0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min', abs=True)
+Male      0.4406
+Female    0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall')
+Male     -0.2436
+Female    0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall', abs=True)
+Male      0.2436
+Female    0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='group', group='Female', abs=True)
+Male      0.4406
+Female    0.0
+Name: TBD dtype: float64
+```
+The arguments introduced so far for the `differences()` method:
+- `relative_to=` to decide the common point for the differences. Possible values are `'max'` (the default), `'min'`, `'overall'` and `'group'`
+- `group=` to select a group name, only when `relative_to` is set to `'group'`. Default is `None`
+- `abs` to indicate whether to take the absolute value of each entry (defaults to false)
+
+The user could then use the Pandas methods `max()` and `min()` to reduce these Series objects to scalars.
+However, this will run into issues where the `relative_to` argument ends up pointing to either the maximum or minimum group, which will have a difference of zero.
+That could then be the maximum or minimum value of the set of difference, but probably won't be what the user wants.
+
+To address this case, we should add an extra argument `aggregate=` to the `differences()` method:
+```python
+>>> result.differences(aggregate='max')
+0.4406
+>>> result.differences(relative_to='overall', aggregate='max')
+0.1870
+>>> result.differences(relative_to='overall', abs=True, aggregate='max')
+0.2436
```

Comment:
why is this different from the line above with no `abs`? The value was positive, so that should be identical, right?

Reply:
Without `abs` the values were -0.2436 and 0.1870. With `abs` the values are 0.2436 and 0.1870.